connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -115,7 +115,7 @@ class SparkSession private[sql] (
   private def createDataset[T](encoder: AgnosticEncoder[T], data: Iterator[T]): Dataset[T] = {
     newDataset(encoder) { builder =>
       val localRelationBuilder = builder.getLocalRelationBuilder
-        .setSchema(encoder.schema.catalogString)
```

Comment:
O damn. This should have been toDDL. This is also wrong in PlanGenerationTestSuite. This also why don't like to work with strings. See: https://github.com/apache/spark/pull/40238

Reply:
Should I change `json` to `toDDL`, or is `json` actually OK?

Reply:
json is ok